src/main/java/com/github/aloomaio/androidsdk/aloomametrics/AnalyticsMessages.java

```diff
@@ -321,61 +354,56 @@ else if (msg.what == KILL_WORKER) {
             }// handleMessage
 
             private void sendAllData(ADbAdapter dbAdapter) {
-                final ServerMessage poster = getPoster();
+                final HttpService poster = getPoster();
                 if (! poster.isOnline(mContext)) {
                     logAboutMessageToAlooma("Not flushing data to alooma because the device is not connected to the internet.");
                     return;
                 }
 
                 logAboutMessageToAlooma("Sending records to alooma");
-                sendData(dbAdapter, ADbAdapter.Table.EVENTS, new String[]{ mSchema + "://" + mAloomaHost + "/track?ip=1" });
+                sendData(dbAdapter, ADbAdapter.Table.EVENTS,
+                        mSchema + "://" + mAloomaHost + "/track?ip=1", mHeaders, mContentType);
             }
 
-            private void sendData(ADbAdapter dbAdapter, ADbAdapter.Table table, String[] urls) {
-                final ServerMessage poster = getPoster();
+            private void sendData(ADbAdapter dbAdapter, ADbAdapter.Table table, String url,
+                                  Map<String, String> headers, RemoteService.ContentType contentType) {
+                final HttpService poster = getPoster();
                 final String[] eventsData = dbAdapter.generateDataString(table);
 
                 if (eventsData != null) {
                     final String lastId = eventsData[0];
                     final String rawMessage = eventsData[1];
 
-                    final String encodedData = Base64Coder.encodeString(rawMessage);
```

Comment:
just wondering if we should leave backwards compatibility

Reply:
Yeah I just moved some of this logic into `HttpService`